model/telemetry/registry.yaml

```diff
@@ -81,3 +81,42 @@ groups:
         brief: >
           The version string of the auto instrumentation agent or distribution, if used.
         examples: ["1.2.3"]
+      - id: telemetry.sdk.processor.type
+        type: string
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK processor.
+        note: |
+          The value `batch-span` MUST be used for the SDK builtin Batch Span Processor.
+        examples: ["batch-span", "MyCustomProcessor"]
+      - id: telemetry.sdk.exporter.type
+        type: string
+        stability: experimental
+        brief: >
+          A name identifying the type of the OpenTelemetry SDK exporter.
+        note: |
+          For OTLP, the values `otlp-grpc`, `otlp-http` or `otlp-http-json` MUST be used, based on the transport and serialization format.
+        examples: ["otlp-grpc", "jaeger"]
+      - id: telemetry.sdk.component.id
+        type: string
+        stability: experimental
+        brief: >
+          A name uniquely identifying the instance of the OpenTelemetry SDK component within its containing SDK instance.
+        note: |
+          The SDK MAY allow users to provide an id for the component instances. If no id is provided by the user,
+          the SDK SHOULD automatically assign an id. Because this attribute is used in metrics, the SDK MUST ensure a low cardinality in that case.
+          E.g. it MUST NOT use a UUID.
+          It instead MAY do that by using the following pattern as value: `<telemetry.sdk.processor/exporter.type>-<instance-counter>`:
+          Hereby, `<instance-counter>` is a monotonically increasing counter (starting with `0`), which is incremented every time an
+          instance of the given component type is created.
+          For example, the first Batch Span Processor will have `batch-span-0` as `telemetry.sdk.component.id`, the second one `batch-span-1` and so one.
+          These values will therefore be reused in the case of an application restart.
+        examples: ["batch-span-0", "custom-name"]
+      - id: telemetry.sdk.failure_reason
```

Comment:
How about reusing [`error.type`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/error.md) for this in the corresponding metrics instead of introducing this new attribute?

As you can see in the definition of `error.type` it's also meant to be low cardinality within an instrumentation.

Reply:
Make sense, done in c24bccc. However, unfortunately I don't see any way of adding new enum members on attribute usage. I would want to enforce that `queue full` MUST be used by Batch Span Processors for spans dropped due to a full queue.
Any suggestions on how to best do this?

Reply:
@lmolkova we don't have a way to extend enum types (when referencing them), yet, right?

Reply:
In [0e54b96](https://github.com/open-telemetry/semantic-conventions/pull/1631/commits/0e54b96ac97f2c6ab2f366c84236a3ec25a9e329) I've for now defined this by overriding the `brief` section for the attribute usage within the metric.